Build/validate-qir.ps1

```diff
@@ -89,35 +104,71 @@ $qirProjects = @(
 
     @{ Path = (Join-Path $PSScriptRoot .. samples numerics CustomModAdd CustomModAdd.csproj); Args = @() },
-    @{ Path = (Join-Path $PSScriptRoot .. samples numerics EvaluatingFunctions EvaluatingFunctions.csproj); Args = @() },
-    @{ Path = (Join-Path $PSScriptRoot .. samples numerics ResourceCounting ResourceCounting.csproj); Args = @() },
+    # memory leak #@{ Path = (Join-Path $PSScriptRoot .. samples numerics EvaluatingFunctions EvaluatingFunctions.csproj); Args = @() },
+    # memory leak #@{ Path = (Join-Path $PSScriptRoot .. samples numerics ResourceCounting ResourceCounting.csproj); Args = @() },
 
     # not compatible #@{ Path = (Join-Path $PSScriptRoot .. samples simulation h2 command-line H2SimulationSampleCmdLine.csproj); Args = @() },
-    @{ Path = (Join-Path $PSScriptRoot .. samples simulation hubbard HubbardSimulationSample.csproj); Args = @() }
+    @{ Path = (Join-Path $PSScriptRoot .. samples simulation hubbard HubbardSimulationSample.csproj); Args = @() },
     # not compatible #@{ Path = (Join-Path $PSScriptRoot .. samples simulation ising adiabatic AdiabaticIsingSample.csproj); Args = @() },
     # not compatible #@{ Path = (Join-Path $PSScriptRoot .. samples simulation ising generators IsingGeneratorsSample.csproj); Args = @() },
     # not compatible #@{ Path = (Join-Path $PSScriptRoot .. samples simulation ising phase-estimation IsingPhaseEstimationSample.csproj); Args = @() },
     # not compatible #@{ Path = (Join-Path $PSScriptRoot .. samples simulation ising simple SimpleIsingSample.csproj); Args = @() },
     # not compatible #@{ Path = (Join-Path $PSScriptRoot .. samples simulation ising trotter-evolution IsingTrotterSample.csproj); Args = @() },
-    # needs argument(s) #@{ Path = (Join-Path $PSScriptRoot .. samples simulation qaoa QAOA.csproj); Args = @() }
+    @{ Path = (Join-Path $PSScriptRoot .. samples simulation qaoa QAOA.csproj); Args = @("--numTrials", "20") }
 )
 
+Write-Host "##[info]Beginning Microsoft.Quantum.Qir.CommandLineTool installation."
 if (-not (Get-Command qir-cli -ErrorAction SilentlyContinue)) {
-    Write-Host "##[error]The qir-cli command is missing; you can install it by running `dotnet tool install Microsoft.Quantum.Qir.CommandLineTool -g`.";
-    $script:allOk = $False
+    $version = ($Env:NUGET_VERSION -split "-")[0] + "-alpha"
+    Write-Host "##[info]Going to install Microsoft.Quantum.Qir.CommandLineTool.$version."
+    dotnet tool install Microsoft.Quantum.Qir.CommandLineTool --version $version -g
+    if  ($LastExitCode -ne 0) {
+        Write-Host "##[error]Failed to install Microsoft.Quantum.Qir.CommandLineTool.$version."
+        $script:allOk = $False
+    } else {
+        Write-Host "##[info]The Microsoft.Quantum.Qir.CommandLineTool.$version has been installed."
+    }
 } else {
+    Write-Host "##[info]The Microsoft.Quantum.Qir.CommandLineTool is already installed."
+}
+
+if ($allOk) {
+    if ($IsWindows) {
```

Comment:
Note that `$IsWindows` is not set on Windows PowerShell (that is, versions 5.1 and earlier). In general, when testing `$IsWindows`, it's helpful to also look at the version table in order to support older versions.
```suggestion
    if ($IsWindows -or $PSVersionTable.PSEdition -eq "Desktop") {
```

Reply:
@swernli had suggested that $IsWindows would be sufficient here. Since I'm not a powershell expert, I'll have to follow the lead of others on this.

Reply:
It absolutely is on PowerShell Core (6.0) and on PowerShell (7.0) or later, but it is not sufficient for Windows PowerShell (5.1 and earlier). In other PRs, you have indicated that you don't want to assume that 6.0 or later is installed. If you do want to take a hard dependency on 6.0 or later, adding the line `#Requires -Edition Core` at the top of the file (below the `#!` line) will enforce that.

https://docs.microsoft.com/powershell/module/microsoft.powershell.core/about/about_requires

Reply:
Making the edit that Chris suggests here makes sense.

Reply:
done